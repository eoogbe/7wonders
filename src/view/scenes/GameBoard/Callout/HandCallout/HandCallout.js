/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import { useSelector } from 'react-redux';
import { getHandCallout } from 'presentation/hand';
import CalloutSection from '../CalloutSection';
import '../Callout.css';

const HandCallout = ({ cardId }) => {
  const { name, type, effects, cost, payment, chainedStructures } =
    useSelector(getHandCallout({ cardId }));

  return (
    <section>
      <h2 className="Callout-title">{name}</h2>
      <p className="Callout-paragraph">Type: {type}</p>
      <CalloutSection heading="This card awards" items={effects} />
      {cost.length > 0 && <CalloutSection heading="It costs" items={cost} />}
      <p className="Callout-paragraph">{payment}</p>
      {chainedStructures.length > 0 &&
        <CalloutSection
          heading="If you have this card, you can build for free"
          items={chainedStructures}
        />}
    </section>
  );
};

export default HandCallout;
