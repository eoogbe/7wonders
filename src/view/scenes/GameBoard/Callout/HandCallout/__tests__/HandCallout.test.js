/**
 * Copyright 2022 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import PLAYERS from '__fixtures__/players';
import HandCallout from '../HandCallout';

const mockStore = configureStore();

it('renders without crashing', () => {
  const store = mockStore(createState());
  const tree =
    renderer.create(
      <Provider store={store}>
        <HandCallout cardId={'LUMBER_YARD'} />
      </Provider>
    ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders with a cost', () => {
  const store = mockStore(createState());
  const tree =
    renderer.create(
      <Provider store={store}>
        <HandCallout cardId={'MINE'} />
      </Provider>
    ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders with chained structures', () => {
  const store = mockStore(createState());
  const tree =
    renderer.create(
      <Provider store={store}>
        <HandCallout cardId={'APOTHECARY'} />
      </Provider>
    ).toJSON();
  expect(tree).toMatchSnapshot();
});

const createState = () => ({
  app: { currentPlayer: 'PLAYER1' },
  players: PLAYERS,
});
