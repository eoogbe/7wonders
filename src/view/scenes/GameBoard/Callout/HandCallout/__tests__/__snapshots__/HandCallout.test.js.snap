// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with a cost 1`] = `
<section>
  <h2
    className="Callout-title"
  >
    Mine
  </h2>
  <p
    className="Callout-paragraph"
  >
    Type: 
    Raw material
  </p>
  <section
    className="CalloutSection"
  >
    <h3
      className="CalloutSection-heading"
    >
      This card awards
      :
    </h3>
    <ul
      className="CalloutSection-list"
    >
      <li>
        1 Ore or Stone
      </li>
    </ul>
  </section>
  <section
    className="CalloutSection"
  >
    <h3
      className="CalloutSection-heading"
    >
      It costs
      :
    </h3>
    <ul
      className="CalloutSection-list"
    >
      <li>
        1 coin
      </li>
    </ul>
  </section>
  <p
    className="Callout-paragraph"
  >
    It will cost 1 coin to build this structure.
  </p>
</section>
`;

exports[`renders with chained structures 1`] = `
<section>
  <h2
    className="Callout-title"
  >
    Apothecary
  </h2>
  <p
    className="Callout-paragraph"
  >
    Type: 
    Science
  </p>
  <section
    className="CalloutSection"
  >
    <h3
      className="CalloutSection-heading"
    >
      This card awards
      :
    </h3>
    <ul
      className="CalloutSection-list"
    >
      <li>
        Compass
      </li>
    </ul>
  </section>
  <section
    className="CalloutSection"
  >
    <h3
      className="CalloutSection-heading"
    >
      It costs
      :
    </h3>
    <ul
      className="CalloutSection-list"
    >
      <li>
        1 Loom
      </li>
    </ul>
  </section>
  <p
    className="Callout-paragraph"
  >
    You cannot afford to build this structure.
  </p>
  <section
    className="CalloutSection"
  >
    <h3
      className="CalloutSection-heading"
    >
      If you have this card, you can build for free
      :
    </h3>
    <ul
      className="CalloutSection-list"
    >
      <li>
        Dispensary
      </li>
      <li>
        Stables
      </li>
    </ul>
  </section>
</section>
`;

exports[`renders without crashing 1`] = `
<section>
  <h2
    className="Callout-title"
  >
    Lumber yard
  </h2>
  <p
    className="Callout-paragraph"
  >
    Type: 
    Raw material
  </p>
  <section
    className="CalloutSection"
  >
    <h3
      className="CalloutSection-heading"
    >
      This card awards
      :
    </h3>
    <ul
      className="CalloutSection-list"
    >
      <li>
        1 Wood
      </li>
    </ul>
  </section>
  <p
    className="Callout-paragraph"
  >
    You can build this structure for free.
  </p>
</section>
`;
